pkg/operator/resource_reconciler.go

```diff
@@ -457,3 +491,16 @@ func ListMatchingNamespaces(selector labels.Selector, nsInf cache.SharedIndexInf
 	}
 	return ns, nil
 }
+
+// hasOwnership returns true if the controller is the "owner" of the object.
+// Whether it's owner is determined by the value of 'controllerID'
+// annotation. If the value matches the controllerID then it owns it, or if the
+// controllerID is "" and there's no annotation set or the annotation value is empty.
+func (rr *ResourceReconciler) hasOwnership(obj metav1.Object) bool {
+	if owner, ok := obj.GetAnnotations()[controllerIDLabel]; ok {
+		return owner == rr.controllerID
+	}
+
+	level.Debug(rr.logger).Log("msg", "skipping object not managed by the controller", "object", fmt.Sprintf("%s/%s", obj.GetNamespace(), obj.GetName()), "object_controller_id", obj.GetAnnotations()[controllerIDLabel], "controller_id", rr.controllerID)
+	return false
```

Comment:
(suggestion) just in case, it's better to guard against nil map.

```suggestion
	var controllerID string
	if obj.GetAnnotations() != nil {
		controllerID = obj.GetAnnotations()[controllerIDLabel]
	}
	if controllerID != rr.controllerID {
		level.Debug(rr.logger).Log("msg", "skipping object not managed by the controller", "object", fmt.Sprintf("%s/%s", obj.GetNamespace(), obj.GetName()), "object_controller_id", controllerID, "controller_id", rr.controllerID)
		return false
	}

	return true
```